pkgs/build-support/flutter/default.nix

```
@@ -223,20 +227,25 @@ let
 
     # binaries need to be patched
     autoPatchelf -- "$depsFolder"
-
-    runHook postConfigure
   '';
 
-  buildPhase = ''
-    runHook preBuild
+  preConfigurePhases = [ "flutterConfigurePhase" ];
 
+  # avoid cmake phase
+  configurePhase = "true";
+
+  flutterBuildPhase = ''
     # for some reason fluffychat build breaks without this - seems file gets overriden by some tool
     mv pubspec-backup pubspec.yaml
     mkdir -p build/flutter_assets/fonts
 
     flutter packages get --offline -v
     flutter build linux --release -v
+  '';
 
+  buildPhase = ''
+    runHook preBuild
+    runHook flutterBuildPhase
     runHook postBuild
   '';
```

What is this? Why not just use buildPhase with preHooks? Looks to me a lot easier overhaul.

---

How do you mean with prehooks? The idea is the user is able to overwrite buildPhase but can optionally call the original one somewhere in the middle of it. What would be the best way to accomplish this?

---

If you want to run something before you would put it into preBuild and for after into postBuild. Basically this would already achieve this:

```suggestion
  preConfigurePhases = [ "flutterConfigurePhase" ];

  # avoid cmake phase
  configurePhase = "true";

  buildPhase = ''
    runHook preBuild

    # for some reason fluffychat build breaks without this - seems file gets overriden by some tool
    mv pubspec-backup pubspec.yaml
    mkdir -p build/flutter_assets/fonts

    flutter packages get --offline -v
    flutter build linux --release -v

    runHook postBuild
  '';
```

---

Hrm I'll leave the phases as-is for now. It was initially added to package appflowy, but that is already packaged as bin derivation.